### PR TITLE
Issue 859

### LINKED ISSUE #859: Definitional Equality in Let Expression
The following doesn't work:

```aya
def let-definitional-equality =>
  let
  | x : Nat := 1
  | x=1 : x = 1 := idp
  in 0
```

<details>
<summary> Aya said: </summary>

```plain
In file LetExpr.aya:54:19 ->

  52 |     let
  53 |     | x : Nat := 1
  54 |     | x=1 : x = 1 := idp
     |                      ^-^

Error: The boundary
         x
       disagrees with
         1
```

</details>

### PR DESCRIPTION
**Pull request by HoshinoTented**
resolve #859.

### CODE REVIEW

File: base/src/main/java/org/aya/tyck/ExprTycker.java [hunk 1]
```diff
@@ -361,7 +346,7 @@ public <T> T withSubSubst(Supplier<T> supplier) {
         var nameAndType = new Term.Param(let.bind().bindName(), definedAsResult.type(), true);
 
         var bodyResult = subscoped(() -> {
-          localCtx.put(nameAndType);
+          definitionEqualities.addDirectly(nameAndType.ref(), definedAsResult.wellTyped(), definedAsResult.type());
```
Comment (HoshinoTented) [inline base/src/main/java/org/aya/tyck/ExprTycker.java:349]: Should we add it to localCtx ?

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/ExprTycker.java:349]: I think it's fine

**Review by ice1000 — APPROVED** [review]
⚗️

**Pull request merged**